taipy/core/_manager/_manager_factory.py

```diff
@@ -0,0 +1,74 @@
+# Copyright 2022 Avaiga Private Limited
+#
+# Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in compliance with
+# the License. You may obtain a copy of the License at
+#
+#        http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software distributed under the License is distributed on
+# an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for the
+# specific language governing permissions and limitations under the License.
+
+from importlib import util
+from typing import Type
+
+from taipy.core.common._utils import _load_fct
+from taipy.core.cycle._cycle_manager import _CycleManager
+from taipy.core.data._data_manager import _DataManager
+from taipy.core.job._job_manager import _JobManager
+from taipy.core.pipeline._pipeline_manager import _PipelineManager
+from taipy.core.scenario._scenario_manager import _ScenarioManager
+from taipy.core.task._task_manager import _TaskManager
+
+
+class _ManagerFactory:
+    _TAIPY_ENTERPRISE_MODULE = "taipy.enterprise"
+    _TAIPY_ENTERPRISE_CORE_MODULE = _TAIPY_ENTERPRISE_MODULE + ".core"
+
+    @classmethod
+    def _scenario_manager(cls) -> Type[_ScenarioManager]:
+        if cls._has_enterprise():
```

Comment:
@jrobinAV should we also include if the config mode stated to use the enterprise module?

Reply:
That would require a new config value in the global app. What do you think? @jrobinAV  @toan-quach

Reply:
yep, the reason I'm asking is in the use case where enterprise users dont want to have authorisation, how should we approach it 😄

Reply:
I guess, if the user WANTS to use authentication and the rest of it, a configuration of how to access the authentication system will have to be provided.
If no such config is provided.... then there will be no authentication...

Reply:
Yes. I believe we will need to have a test on the config at a point. This is not blocking the PR anyway.